Swiftz/Reader.swift

```diff
@@ -11,15 +11,10 @@ public struct Reader<R, A> {
     /// The function that modifies the environment
     public let reader : R -> A
     
-    public init(_ reader : R -> A) {
+    init(_ reader : R -> A) {
```

Comment:
I thought hard about what it would mean to leave the constructor public and I didn't see a reason to keep it around in the presence of the actual `reader` function, which calls through to this as an implementation detail.  In the future (not saying we should ever do this) should we need to alter the definition of reader's internals, the initializer can be altered without fear of leaking API details.